nomad/plan_apply.go

```diff
@@ -496,6 +504,12 @@ func evaluatePlan(pool *EvaluatePool, snap *state.StateSnapshot, plan *structs.P
 		return &structs.PlanResult{RefreshIndex: index}, nil
 	}
 
+	// Check to ensure this job plan does not create duplicate allocation
+	// indexes within a job version.
+	if err := evaluatePlanAllocIndexes(snap, plan); err != nil {
+		return nil, err
+	}
```

Comment:
`system` job (and `sysbatch`?) allocations always have alloc index 0, so we need to have a conditional around job type before calling this function.

Reply:
Correct on system, however, sysbatch are unique per set of allocations on a node. This will therefore require followup work. I will add the conditional logic to skip these two, and then followup with another PR to handle sysbatch job types.

Reply:
I have resolved this within: https://github.com/hashicorp/nomad/pull/18376/commits/729421a1ee456e8ece4dc91f9b187db5788ce677

I added this within the function, so the logic is within this single place. I think this will make it easier to maintain, particularly with the future need for another decision tree for sysbatch.